tests/test_examples.py

```diff
@@ -0,0 +1,390 @@
+    KEEP_ARTIFACTS_PATH = Path(artifacts_path)
+    KEEP_ARTIFACTS_PATH.mkdir(parents=True, exist_ok=True)
+else:
+    KEEP_ARTIFACTS_PATH = None
+
+
+def _execute(
+    cmd: Iterable[str], installer_input=None, check=True, **env_vars
+) -> subprocess.CompletedProcess:
+    t0 = time.time()
+    if env_vars:
+        env = os.environ.copy()
+        env.update({k: v for (k, v) in env_vars.items() if v is not None})
+    else:
+        env = None
+    p = subprocess.Popen(
+        cmd,
+        stdin=subprocess.PIPE if installer_input else None,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+        env=env,
+    )
+    stdout, stderr = None, None
+    try:
+        stdout, stderr = p.communicate(input=installer_input, timeout=420)
+        retcode = p.poll()
+        if check and retcode:
+            raise subprocess.CalledProcessError(retcode, cmd, output=stdout, stderr=stderr)
+        return subprocess.CompletedProcess(cmd, retcode, stdout, stderr)
+    except subprocess.TimeoutExpired:
+        p.kill()
+        stdout, stderr = p.communicate()
+        raise
+    finally:
+        if stdout:
+            print(stdout)
+        if stderr:
+            print(stderr, file=sys.stderr)
+        print("Took", timedelta(seconds=time.time() - t0))
+
+
+def _run_installer_exe(installer, install_dir, installer_input=None):
+    """
+    NSIS manual:
+    > /D sets the default installation directory ($INSTDIR), overriding InstallDir
+    > and InstallDirRegKey. It must be the last parameter used in the command line
+    > and must not contain any quotes, even if the path contains spaces. Only
+    > absolute paths are supported.
+    Since subprocess.Popen WILL escape the spaces with quotes, we need to provide
+    them as separate arguments. We don't care about multiple spaces collapsing into
+    one, since the point is to just have spaces in the installation path -- one
+    would be enough too :)
+    This is why we have this weird .split() thingy down there in `/D=...`.
+    """
+    if not sys.platform.startswith("win"):
+        raise ValueError("Can only run .exe installers on Windows")
+    if "NSIS_USING_LOG_BUILD" not in os.environ:
+        warnings.warn(
+            "Windows installers are tested with NSIS in silent mode, which does "
+            "not report errors on exit. You should use logging-enabled NSIS builds "
+            "to generate an 'install.log' file this script will search for errors "
+            "after completion."
+        )
+    cmd = ["cmd.exe", "/c", "start", "/wait", installer, "/S", *f"/D={install_dir}".split()]
+    _execute(cmd)
+
+    # Windows installers won't raise exit codes so we need to check the log file
+    error_lines = []
+    try:
+        log_is_empty = True
+        with open(os.path.join(install_dir, "install.log"), encoding="utf-16-le") as f:
+            for line in f:
+                log_is_empty = False
+                if ":error:" in line:
```

Comment:
Do we want to use `line.lower()` to make ourselves case-independent?

Reply:
```suggestion
                if ":error:" in line.lower():
```

Reply:
Done, thanks! Unfortunately I'll need the approval again, as it gets automatically dismissed with new changes 😬